xl/trax/util.py

```diff
@@ -103,7 +103,7 @@ def get_tracks_from_uri(uri):
 def sort_tracks(
     fields: Iterable[str],
     items: Iterable[T],
-    trackfunc: Optional[Callable[[T], Track]] = None,
+    trackfunc: Optional[Callable[[Track], Track]] = None,
```

Comment:
This doesn't seem right. From my understanding, `tracfunc` is meant to transform some unknown type `T` (element of `items`) into a `Track`. That is, it's a function with this signature:

    def trackfunc(item: T) -> Track:
        pass

I'll try checking with mypy later.

Reply:
OK, I've tested this and I think the existing annotation is correct (`Callable[[T], Track]`), but the body needs this change:

```py
if trackfunc is None:
    trackfunc = lambda tr: cast(Track, tr)
```

That is:

* If `trackfunc` is specified, `T` can be anything. `trackfunc` then transforms this `T` into a `Track`.
* If `trackfunc` is not specified, `T` must already be `Track`.

I don't know if this can be expressed easily in the type signature, but I think for now the cast is good enough.